mattermost/mattermost-webapp
# E2E: Initial test setup for reusable component

#### Summary
This PR aims to provide proof of concept on how to test reusable components.

It test how the regular badge renders:
- UI check via text and CSS properties
- text change
- show/hide

How to run the test:
1. Run storybook by `npm run storybook` and see that it's running via `localhost:6006`
2. Run Cypress by `cd e2e`, then `npm run cypress:open`
3. Select and run test via "Integration Tests > widgets > badges > regular_spec.js"
4. See that all tests passed.  Tests include:
- verify UI
- verify text change
- verify hide and show

#### Ticket Link
Jira ticket: [MM-19657](https://mattermost.atlassian.net/browse/MM-19657)

~~(Note this is submitted on top https://github.com/mattermost/mattermost-webapp/pull/4024.  Will update once it's merged.)~~ Said PR is merged and this PR is now updated.
